src/luarocks/fs/lua.lua

```diff
@@ -544,86 +544,183 @@ end
 if socket_ok then
 
 local ltn12 = require("ltn12")
+local http = require("socket.http")
 local luasec_ok, https = pcall(require, "ssl.https")
 
-local redirect_protocols = {
-   http = http,
-   https = luasec_ok and https,
-}
+-- Patch ltn12 source used by LuaSocket to get fixed length
```

Comment:
It is very stable (underlying code not changed since 2004). Tested it with the oldest version of Luasocket available on Luarocks (2.0.1-2), it works.

What do you mean by being not responsive? It updates too rarely?